mesh/mesh.go

```diff
@@ -50,11 +50,16 @@ type AtxDB interface {
 	SyntacticallyValidateAtx(atx *types.ActivationTx) error
 }
 
+type TxMempool interface {
```

Comment:
tl;dr: `TxMempool` is a confusing name and I'll change it.

It's not the same type underneath - this interface is implemented only by `BlockBuilder` since it has access to both the `TransactionProcessor` (as a validator) and the mempool. I'll find a better name to eliminate the confusion.